games/TicTacToe/src/utils.sw

```diff
@@ -0,0 +1,68 @@
+library utils;
+
+use core::ops::Eq;
+
+impl<T> Eq for Option<T> {
+    fn eq(self, other: Self) -> bool {
+        match (self, other) {
+            (Option::None, Option::None) => true,
+            (Option::Some(T), Option::Some(T)) => true,
+            _ => false,
+        }
+    }
+}
+
+pub fn win_check(board: Vec<Option<Identity>>, player: Option<Identity>) -> bool {
+    let matches = [
```

Comment:
I would add a comment here explaining the purpose of this variable.  I understand it is all of the possible win conditions, but it is not immediately apparent.

Reply:
We have considered factoring out the array into storage and using it as a `read-only` value; however, storage does not support regular arrays.
Alternatively, we've considered moving it into a function and using that as a getting in the utility functions.